cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -405,4 +347,108 @@ public synchronized String toString() {
                                                        ? "None" : _ongoingSelfHealingAnomaly.anomalyId(),
                          BALANCEDNESS_SCORE, _balancednessScore);
   }
+
+  @JsonResponseClass
+  @JsonResponseExternalFields(KafkaAnomalyType.class)
+  protected static class SelfHealingEnabledRatio {
+    private Map<String, Float> _selfHealingEnabledRatioMap = new HashMap<>(KafkaAnomalyType.cachedValues().size());
+
+    SelfHealingEnabledRatio(int size) {
+      _selfHealingEnabledRatioMap = new HashMap<>(size);
+    }
+
+    public void put(AnomalyType anomalyType, Float value) {
+      _selfHealingEnabledRatioMap.put(anomalyType.toString(), value);
+    }
+
+    protected Map<String, Float> getJsonStructure() {
+      return _selfHealingEnabledRatioMap;
+    }
+  }
+
+  @JsonResponseClass
+  protected static class AnomalyDetails {
+    @JsonResponseField(required = false)
+    private static final String STATUS_UPDATE_MS = "statusUpdateMs";
```

Comment:
`STATUS_UPDATE_MS ` and `DETECTION_MS ` are required JSON field?
`STATUS_UPDATE_DATE ` and `DETECTION_DATE ` are not JSON field so should not be annotated?

Reply:
Ah I think  I confused myself by changing the function name to `getJsonStructure()` when it doesn't only deal with the json response. I changed the function name backed and made them required responses for the json response, as you pointed out.